### DISCUSSION TITLE
SWC config in Next.js 12

### DISCUSSION THREAD

**Discussion by trm217** [vercel/next.js#36379] [Help] [answered]
I am having some troubles getting our Next.js App to compile so that it only contains es5 syntax to cause no syntax errors in ie11. (As of now after switching to SWC, arrow functions are still present in the static JS-files for the website and thus ie11 crashes)

We used to have the following config inside our .babelrc file that helped in this case
```json
{
  …
  "env": {
    "production": {
      "presets": [
        [
          "next/babel",
          {
            "preset-env": {
              "targets": "defaults, IE 11, iOS 9"
            }
          }
        ]
      ]
    },
    …
}
```
  
Now with SWC from what I read in the [SWC docs](https://swc.rs/docs/configuration/supported-browsers#targets), it should work fairly similar using the env targets property in the .swcrc config-file. So I setup my config as follows:
```json
{
  "env": {
    "targets": "defaults, IE 11, iOS 9"
  }
}
```
Now the problem I am having the Next.js output doesn't change at all based on this config. I additionally tried setting the compile target to es5, but that didn't work neither.

The Next.js doc still is very lacking on how one can configure SWC inside a next.js project. Since the equivalent is still present for babel however, I think this should really be added. ~@kdy1 In case you could point me in the right direction, I would be glad to contribute into extending the docs on this.~

**Comment by kdy1** [comment, marked as answer]
`.swcrc` is not supported at the moment

**Reply by trm217** [reply]
Oh, so what is the solution in regards to compiling for specific browser versions?

**Reply by FelipeLahti** [reply]
Any plans on supporting `.swcrc` @kdy1 ? I'm looking to compile to commonjs as cypress component tests today only support stubbing commonjs and not esm. As an workaround we're using babel but we would love to use swc.
